src/main/java/org/zanata/mt/model/TextFlow.java

```diff
@@ -0,0 +1,98 @@
+package org.zanata.mt.model;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+import javax.persistence.Access;
+import javax.persistence.AccessType;
+import javax.persistence.CascadeType;
+import javax.persistence.Entity;
+import javax.persistence.FetchType;
+import javax.persistence.JoinColumn;
+import javax.persistence.ManyToOne;
+import javax.persistence.OneToMany;
+import javax.persistence.Transient;
+import javax.validation.constraints.Size;
+
+import org.hibernate.annotations.NaturalId;
+import org.hibernate.validator.constraints.NotEmpty;
+import org.zanata.mt.api.dto.LocaleId;
+import org.zanata.mt.util.TranslationUtil;
+
+/**
+ * @author Alex Eng <a href="mailto:aeng@redhat.com">aeng@redhat.com</a>
+ */
+@Entity
+@Access(AccessType.FIELD)
+public class TextFlow extends ModelEntity {
```

Comment:
We might want to call it something other than TextFlow to avoid confusion with the Text Flow in other projects.